jit/v2/JIT_v2.0.0.yaml

```diff
@@ -940,6 +936,263 @@ components:
+    universalExportVoyageReferenceQueryParam:
+      in: query
+      name: universalExportVoyageReference
+      description: |
+        The **Universal Export Voyage Reference** to filter by. The result will only return **Port Call Service** objects including the `universalExportVoyageReference`.
+      schema:
+        type: string
+        pattern: ^\d{2}[0-9A-Z]{2}[NEWSR]$
+        minLength: 5
+        maxLength: 5
+        example: 2103N
+
+    # Port Call Service query parameters
+
+    portCallServiceTypeQueryParam:
+      in: query
+      name: portCallServiceType
+      description: |
+        The **Port Call Service Type** to filter by. The result will only return **Port Call Service** objects including the `portCallServiceType`.
+      schema:
+        type: string
+        maxLength: 50
+        pattern: ^\S(?:.*\S)?$
+        example: BERTH
+    portCallServiceEventTypeCodeQueryParam:
+      in: query
+      name: portCallServiceEventTypeCode
+      description: |
+        The **Port Call Service Event Type Code** to filter by. The result will only return **Port Call Service** objects including the `portCallServiceEventTypeCode`.
+      schema:
+        type: string
+        enum:
+          - STRT
+          - CMPL
+          - ARRI
+          - DEPA
+        example: STRT
+
+    # Vessel query parameters
+
+    vesselIMONumberQueryParam:
+      in: query
+      name: vesselIMONumber
+      description: |
+        The **Vessel IMO Number** to filter by. The result will only return **Port Call Service** objects including the `vesselIMONumber`.
+      schema:
+        type: string
+        pattern: ^\d{7,8}$
+        minLength: 7
+        maxLength: 8
+        example: '9321483'
+    vesselNameQueryParam:
+      in: query
+      name: vesselName
+      description: |
+        The **Vessel Name** to filter by. The result will only return **Port Call Service** objects including the `vesselName`.
+      schema:
+        type: string
+        pattern: ^\S(?:.*\S)?$
+        maxLength: 50
+        example: King of the Seas
+    MMSINumberQueryParam:
+      in: query
+      name: MMSINumber
+      description: |
+        The **MMSI Number** to filter by. The result will only return **Port Call Service** objects including the `MMSINumber`.
+      schema:
+        type: string
+        pattern: ^\d{9}$
+        minLength: 9
+        maxLength: 9
+        example: '278111222'
+
+    # Pagination query parameters
+
+    limitQueryParam:
+      in: query
+      name: limit
+      description: |
+        Specifies the maximum number of `Port Call Service` objects to return.
+      schema:
+        type: integer
+        format: int32
+        minimum: 1
+        default: 100
+        example: 100
+    cursorQueryParam:
+      in: query
+      name: cursor
+      description: |
+        A server generated value to specify a specific point in a collection result, used for pagination.
+      schema:
+        type: string
+        maxLength: 1024
```

Comment:
Why should the cursor be 1024 chars? That's way more than needed, right?

Reply:
It depends on how you implement it. If it contains a pointer to a row in a database table - then I agree with you.
In the past we have (internally in DCSA) implemented the cursor as an encoded pageSize, pageNr (+ all filter parameters). In this case 1024 might even be too little in the case of JIT